pkg/maps/ctmap/ctmap.go

```diff
@@ -444,3 +495,58 @@ func Flush(m *bpf.Map, mapName string) int {
 		return 0
 	}
 }
+
+// checkAndUpgrade determines whether the ctmap on the filesystem has different
+// map properties than this version of Cilium, and if so, removes the map from
+// the filesystem so that a subsequent BPF prog install will recreate it with
+// the correct map properties.
+//
+// Returns true if the map was upgraded.
+func checkAndUpgrade(m *bpf.Map, e CtEndpoint, keySize uint32) bool {
+	desiredMapInfo := &bpf.MapInfo{
+		MapType:   bpf.MapTypeHash,
```

Comment:
On newer kernels, this is always an LRU. (Affects all versions with this patch)

https://github.com/cilium/cilium/blob/f8ccf6477f80dbe7c6b83ce03840adec3444bd5a/bpf/bpf_lxc.c#L53-L56

I primarily focused on testing with the dev VM, so didn't catch this earlier.

Reply:
PR is out for the issue I highlighted: #5197